src/video.cpp

```diff
@@ -220,7 +313,11 @@ void CVideo::init_window()
 		window_flags |= SDL_WINDOW_MAXIMIZED;
 	}
 
+	// TODO: fix whatever is crashing the rendering context when accelerated
+	// We can force software rendering here,
+	// but if we don't specify anything SDL should choose correctly.
 	uint32_t renderer_flags = SDL_RENDERER_SOFTWARE;
+	//uint32_t renderer_flags = 0;
```

Comment:
Is this line still needed?

Reply:
Yeah - swapping the comment on those lines will enable hardware accelleration for platforms that support it. Ideally this should be enabled, but the line above forces software rendering in all cases.

I tried enabling it but got some crashes with an invalid GL context. It was a bit out of scope for this PR, but definitely should be looked into in the future.

Reply:
Just a further note that i found the thread that was clearing the display and got this working, but i don't want to mix that up with this PR so i'm leaving it be for now.

Reply:
What thread was? Only one thread should handle drawing.

Reply:
The thread that handles the loading bar in `gui/dialogs/loading_screen.cpp` - which is passed a lambda when constructed in `play_controller::init()`. Inside that lambda it creates a `game_display`, and inside the constructor for `game_display` it called `CVideo::clear()` on the global video instance. So it ended up that the loading screen thread was clearing the screen.

When i remove the call to clear the screen from the `game_display` constructor everything seems to work. The call seems unnecessary anyway.

Reply:
I recommend opening a separate PR for that if you're not adding it to this one.

Reply:
Will do, i have a branch set up but it depends on this one so i'm waiting for a little more testing of this before posting it.